File: SpringBoot-SecKill-Scala/src/main/scala/cn/edu/jxnu/seckill/vo/GoodsVo.scala
package cn.edu.jxnu.seckill.vo
import scala.language.implicitConversions
import scala.beans.BeanProperty
import cn.edu.jxnu.seckill.domain.Goods
import java.util.Date

/**
 * 商品视图对象
 *
 * @author 梦境迷离.
 * @time 2018年5月19日
 * @version v1.0
 */
class GoodsVo extends Goods {

    //秒杀价格
    @BeanProperty
    var seckillPrice: Double = _;

    //库存
    @BeanProperty
    var stockCount: Integer = _;

    //开始日期
    @BeanProperty
    var startDate: Date = _;

    //结束日期
    @BeanProperty
    var endDate: Date = _;

}

File: seckill/src/main/java/cn/edu/jxnu/seckill/vo/GoodsVO.java
package cn.edu.jxnu.seckill.vo;

import java.util.Date;

import cn.edu.jxnu.seckill.domain.Goods;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = false)
public class GoodsVO extends Goods {

	private Double seckillPrice;
	private Integer stockCount;
	private Date startTime;
	private Date endTime;

}
